### ISSUE TITLE
No Support for $facet aggregation pipeline stage

### ISSUE THREAD

**Issue by JuergetSTatVGOA** [bwaldvogel/mongo-java-server#91]
When trying to test a aggregation pipeline with a $facet stage, we're getting following error:

> com.mongodb.MongoCommandException: Command failed with error 40324 (Location40324): 'Unrecognized pipeline stage name: '$facet'' on server localhost:51572. The full response is { "$err" : "Unrecognized pipeline stage name: '$facet'", "errmsg" : "Unrecognized pipeline stage name: '$facet'", "code" : 40324, "codeName" : "Location40324", "ok" : 0 }

The Pipeline is build as follows

```
      MatchOperation matchStage = Aggregation
            .match(new Criteria("_id").is(id))
                .and(SORT_FIELD).gte(start).lte(end));

        FacetOperation facetStage = Aggregation.facet(group().count().as("count")).as("pageInfo")
            .and(sort(sortOrder, SORT_FIELD), skip((long) size * (page == 0 ? page : page - 1)), limit(size)).as("records");

        AggregateObject results = mongoTemplate
            .aggregate(newAggregation(matchStage, facetStage), Datapoint.class, AggregateObject.class)
            .getUniqueMappedResult();
```

**Comment by bwaldvogel** [comment]
I’ve started to add support for `$facet` stages in a265b60.
Could you re-test on latest master?

**Comment by JuergetSTatVGOA** [comment]
I've just build and tested the 1.20.0-SNAPSHOT version and at least in my scenario it works like a charm. Thanks for the quick response!

**Comment by bwaldvogel** [comment]
The change is released in version `1.20.0`.
